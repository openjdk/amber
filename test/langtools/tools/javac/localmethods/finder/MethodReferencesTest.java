/**
 * @test /nodynamiccopyright/
 * @bug 8233754
 * @compile/fail/ref=MethodReferencesTest.out -Werror -XDrawDiagnostics -XDfind=local-methods MethodReferencesTest.java
 */

public class MethodReferencesTest {

    private static void run(Runnable r) {} // can be local
    private static void run() {} // cannot be local since a method reference targets

    public static void main(String [] args) {
        run();
        run(MethodReferencesTest::run);
    }
}
