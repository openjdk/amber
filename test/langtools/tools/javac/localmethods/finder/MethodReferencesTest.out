MethodReferencesTest.java:9:25: compiler.warn.method.could.be.declared.local: main(java.lang.String[]), kindname.class, MethodReferencesTest
- compiler.err.warnings.and.werror
1 error
1 warning
